demo/basic/reusability/reusability.component.ts

```diff
@@ -0,0 +1,50 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'reusability-demo',
+  template: `
+    <div>
+      <h3>
+        Reusability
+        <small>
+          <a
+            href="https://github.com/swimlane/ngx-datatable/blob/master/demo/basic/reusability/reusability.component.ts"
+            target="_blank"
+          >
+            Source
+          </a>
+        </small>
+      </h3>
+      <reusable-table [columns]="columns" [rows]="rows">
+        <ngx-datatable-column name="Gender"></ngx-datatable-column>
+        <ngx-datatable-column name="Company"></ngx-datatable-column>
+      </reusable-table>
+    </div>
+  `
+})
+export class ReusabilityComponent {
+  rows = [];
+
```

Comment:
Is it really reusable when you are hard-coding the column values at component level.
The reason i said this because i tried using the same thing but the passing the data via @Input property and it throws error

Reply:
The columns passed via @input need to be of type `QueryList<DataTableColumnDirective>`, I assume it will fail any other way.